File: src/main/java/com/back_servicios/app_cosultas_servicios/domain/entity/Factura.java
package com.back_servicios.app_cosultas_servicios.domain.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "factura")
public class Factura {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "factura_id")
    private Long facturaId;
    @Column(name = "fecha_periodo_inicio")
    private LocalDate fecha_periodoInicio;

    @Column(name = "fechaPeriodoFin")
    private LocalDate fechaPeriodoFin;

    @Column(name = "consumo_total")
    private BigDecimal consumoTotal;
    private BigDecimal costo_total;

    @ManyToOne
    @JoinColumn(name = "hogar_id", nullable = false)
    @JsonIgnore
    private Hogar hogar;

    @ManyToOne
    @JoinColumn(name = "servicio_id", nullable = false)
    @JsonBackReference
    private Servicios servicios;

}

File: Flutter/lib/helpers/model/Factura.dart
class Factura {
  final int facturaId;
  final String fechaInicio;
  final String fechaFin;
  final double consumoTotal;
  final double costoTotal;

  Factura({
    required this.facturaId,
    required this.fechaInicio,
    required this.fechaFin,
    required this.consumoTotal,
    required this.costoTotal,
  });

  factory Factura.fromJson(Map<String, dynamic> json) {
    return Factura(
      facturaId: json['facturaId'] ?? 0,
      fechaInicio: json['fecha_periodoInicio'] ?? '',
      fechaFin: json['fechaPeriodoFin'] ?? '',
      consumoTotal: (json['consumoTotal'] ?? 0).toDouble(),
      costoTotal: (json['costo_total'] ?? 0).toDouble(),
    );
  }
}